flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java

```diff
@@ -213,43 +214,34 @@ public List<String> retrieveTablePartitions(String project, String dataset, Stri
         }
 
         @Override
-        public Optional<Tuple2<String, StandardSQLTypeName>> retrievePartitionColumnName(
+        public Optional<TablePartitionInfo> retrievePartitionColumnInfo(
                 String project, String dataset, String table) {
             try {
-                String query =
-                        Arrays.asList(
-                                        "SELECT",
-                                        "  column_name, data_type",
-                                        "FROM",
-                                        String.format(
-                                                "  `%s.%s.INFORMATION_SCHEMA.COLUMNS`",
-                                                project, dataset),
-                                        "WHERE",
-                                        String.format(" table_catalog = '%s'", project),
-                                        String.format(" AND table_schema = '%s'", dataset),
-                                        String.format(" AND table_name = '%s'", table),
-                                        " AND is_partitioning_column = 'YES';")
-                                .stream()
-                                .collect(Collectors.joining("\n"));
+                Table tableInfo = BigQueryUtils.tableInfo(bigquery, project, dataset, table);
```

Comment:
We only support native BQ tables for now. Created the TODO (noting Spark connector's approach) for highlighting this observation in case we do opt to support external tables.

Thanks for pointing out!